File: backend/src/db/schema/eventLogs.ts
import { InferInsertModel, InferSelectModel } from "drizzle-orm";
import {
  bigserial,
  pgEnum,
  pgTable,
  text,
  timestamp,
  varchar,
} from "drizzle-orm/pg-core";

export const eventTypeEnum = pgEnum("eventType", [
  "info",
  "warning",
  "critical",
]);


type SystemPrefix = "backend" | "advertisementService" | "eventService";

// Use it in the template literal type
type SystemFormat = `${SystemPrefix}:${string}`;

export const eventLogsTable = pgTable("eventLogs", {
  id: bigserial("id", { mode: "number" }).primaryKey(),
  type: eventTypeEnum("type").notNull(),
  system: varchar("system", { length: 255 })
    .notNull()
    .$type<SystemFormat>(),
  message: text("message").notNull(),
  dateTime: timestamp("dateTime", { withTimezone: true }).defaultNow(),
});

export type selectEventLog = InferSelectModel<typeof eventLogsTable>;
export type insertEventLog = InferInsertModel<typeof eventLogsTable>;

File: localServices/videoProcessing/db/schema/eventLogs.py
from datetime import datetime
from enum import Enum
from sqlalchemy import Enum as SQLAlchemyEnum
from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
from sqlalchemy import Text, DateTime


class Base(DeclarativeBase):
    pass


class EventType(str, Enum):
    INFO = "info"
    WARNING = "warning"
    CRITICAL = "critical"


class EventLog(Base):
    __tablename__ = "eventLogs"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    type: Mapped[EventType] = mapped_column(
        SQLAlchemyEnum(EventType, name="eventtype", create_constraint=False),
        nullable=False,
    )
    message: Mapped[str] = mapped_column(Text, nullable=False)
    date_time: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, default=datetime.utcnow
    )

